pkg/generators/models/templates/oneof.gotpl

```diff
@@ -0,0 +1,118 @@
+// This file is auto-generated, DO NOT EDIT.
+//
+// Source:
+//     Title: {{.SpecTitle}}
+//     Version: {{.SpecVersion}}
+package {{ .PackageName }}
+
+import (
+	"encoding/json"
+	"github.com/mitchellh/mapstructure"
+	{{- if .Discriminator.Field }}
+	"fmt"
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+	{{- end}}
+)
+
+{{- $modelName := .Name }}
+{{- $interfaceName := (printf "%ser" .Name) | firstLower}}
+
+{{- if .Discriminator.Field }}
+type {{$interfaceName}} interface {
+	Get{{.Discriminator.Field|toPascalCase}}() string
```

Comment:
I am making a pretty strong assumption here, that the discriminator is always a string. This works in our API specs, but may not work everywhere. I wasn't sure how to get the actual metadata for the field, so we can definitely improve this behavior

Reply:
This wont work in the Hub API, you can see the failure here, the code doesn't compile https://github.com/contiamo/hub/pull/880  We get errors like 

```
pkg/openapi/models/model_connection_test_properties.go:53:10: cannot use data (type AthenaProduct) as type connectionTestPropertieser in assignment:
        AthenaProduct does not implement connectionTestPropertieser (wrong type for GetProductName method)
                have GetProductName() AthenaProductProductName
                want GetProductName() string
```
because each sub-type defines a single valued enum for the discrimination field. As a result, we don't have a unified type that we can use as the interface. We can probably give up the interface, but we need to do a few experiments first to verify.

Reply:
I implemented a workaround in this commit https://github.com/contiamo/openapi-generator-go/pull/57/commits/6de5b7ed5a2a1f91d8c2a47886e23aabe4c73b6c